ts/Stock/Navigator/Navigator.ts

```diff
@@ -598,7 +673,11 @@ class Navigator {
         if (this.hasDragged && !defined(pxMin)) {
             return;
         }
-
+        if (this.isDirty) {
+            // Update DOM navigator elements
+            this.renderElements();
+            this.isDirty = false;
```

Comment:
Are you sure this is the right place to set this flag? Usually,  I would expect flags like this to be set at the end of the `render()` function, not in the middle. In this particular case, it might not be a problem, because we have just one `fireEvent()` here. But with multiple events, we could get different states, even when we are during the rendering (by during rendering I mean methods like `drawMasks()` are yet to be called).